File: find_missing_en.py
import re
import os

files = [
    "lib/visualLab/data/poiExtraSouthsudanCitiesV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanEconomicV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanHistoryV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanLandmarksV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanLifeV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanNatureV2.ts",
    "lib/visualLab/data/poiExtraSouthsudanReliefV2.ts"
]

missing_en = []

for file_path in files:
    if not os.path.exists(file_path):
        continue
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Simple split by POI object start
    pois = content.split('{')
    current_id = None
    for p in pois:
        id_match = re.search(r'id:\s*"(.*?)"', p)
        if id_match:
            current_id = id_match.group(1)
            # Check if descriptionAdvanced exists and has en
            desc_adv_match = re.search(r'descriptionAdvanced:\s*\{', p)
            if desc_adv_match:
                # Find the closing } of descriptionAdvanced
                # This is tricky with regex, but let's look for en: "..." inside
                en_match = re.search(r'en:\s*"[^"]+"', p) # Simplified
                if not en_match or 'en: ""' in p or 'en: ""' in p:
                     missing_en.append(current_id)
            else:
                missing_en.append(current_id)

for mid in missing_en:
    print(mid)

File: find_missing_en.js
import fs from 'fs';
import path from 'path';

const files = [
  "lib/visualLab/data/poiExtraCapeverdeCitiesV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeEconomicV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeHistoryV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeLandmarksV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeLifeV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeNatureV2.ts",
  "lib/visualLab/data/poiExtraCapeverdeReliefV2.ts"
];

const results = [];

files.forEach(file => {
  const filePath = path.join(process.cwd(), file);
  if (!fs.existsSync(filePath)) return;
  const content = fs.readFileSync(filePath, 'utf-8');
  
  // Very crude parser for POI objects
  const poiMatches = content.match(/\{id: "([^"]+)"[\s\S]*?\}/g);
  if (poiMatches) {
    poiMatches.forEach(poiStr => {
      const idMatch = poiStr.match(/id: "([^"]+)"/);
      if (idMatch) {
        const id = idMatch[1];
        const hasEnDescription = poiStr.includes('descriptionAdvanced:') && poiStr.includes('en: "') && poiStr.match(/descriptionAdvanced:\s*\{[^}]*?en:\s*"[^"]+/);
        // More robust check: check if descriptionAdvanced has an 'en' key that is not empty
        const descAdvMatch = poiStr.match(/descriptionAdvanced:\s*\{([\s\S]*?)\}/);
        let missingEn = true;
        if (descAdvMatch) {
          const inner = descAdvMatch[1];
          if (inner.includes('en: "') && !inner.includes('en: ""')) {
            missingEn = false;
          }
        } else {
            // If descriptionAdvanced is missing entirely
            missingEn = true;
        }

        if (missingEn) {
          results.push({ id, file });
        }
      }
    });
  }
});

console.log(JSON.stringify(results, null, 2));
